cernopendata/modules/datacite/cli.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of CERN Open Data Portal.
+# Copyright (C) 2017 CERN.
+#
+# CERN Open Data Portal is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# CERN Open Data Portal is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with CERN Open Data Portal; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+import random
+import string
+
+import click
+from datacite import schema40
+from flask.cli import with_appcontext
+from invenio_pidstore.errors import PIDDoesNotExistError
+from invenio_pidstore.models import PersistentIdentifier
+from invenio_records.api import Record
+
+from .client import DataCiteMDSClientWrapper
+from .providers import DataCiteProviderWrapper
+from .serializers import DataCiteSerializer
+
+
+@click.group()
+def datacite():
+    """Commands for DataCite DOIs management."""
+
+
+@datacite.command()
+@with_appcontext
+def get_registered():
+    """Print all registered DOIs from DataCite."""
+    dcp = DataCiteMDSClientWrapper()
+
+    print(dcp.doi_get_all())
+
+
+@datacite.command()
+@click.option(
+    '--recid',
+    help='Register record with given recid')
+@with_appcontext
+def register(recid):
```

Comment:
Hmm, are we going to mind record IDs for new objects in the future?  Provided we expose DOIs and stuff, we may not even have internal record IDs, only UUIDs...

Reply:
sure, will use UUIDs instead